diff --git a/node_modules/express-session/index.js b/node_modules/express-session/index.js
index d7efeab..3dbb939 100644
--- a/node_modules/express-session/index.js
+++ b/node_modules/express-session/index.js
@@ -77,6 +77,7 @@ var defer = typeof setImmediate === 'function'
  * @param {Boolean} [options.resave] Resave unmodified sessions back to the store
  * @param {Boolean} [options.rolling] Enable/disable rolling session expiration
  * @param {Boolean} [options.saveUninitialized] Save uninitialized sessions to the store
+ * @param {Function} [options.sessionid] Manually set session ID
  * @param {String|Array} [options.secret] Secret for signing session ID
  * @param {Object} [options.store=MemoryStore] Session store
  * @param {String} [options.unset]
@@ -93,6 +94,8 @@ function session(options) {
   // get the session id generate function
   var generateId = opts.genid || generateSessionId
 
+  var getManualSessionId = options.sessionid || function(req) {};
+
   // get the session cookie name
   var name = opts.name || opts.key || 'connect.sid'
 
@@ -214,7 +217,7 @@ function session(options) {
     req.sessionStore = store;
 
     // get the session ID from the cookie
-    var cookieId = req.sessionID = getcookie(req, name, secrets);
+    var cookieId = req.sessionID = getManualSessionId(req) ? getManualSessionId(req) : getcookie(req, name, secrets);
 
     // set-cookie
     onHeaders(res, function(){
